### DISCUSSION TITLE
[line chart] Can I prevent pan from panning into empty space outside the data range?  Can I 'lock' the chart's visible area?

### DISCUSSION THREAD

**Discussion by respectabiggle** [chartjs/chartjs-plugin-zoom#777] [Q&A]
Currently (default behavior), I can pan to the left or right past all my data points, into empty space.  (mode: 'x')

Can I lock/limit/boundary the chart somehow, so that one cannot pan outside of the range of data points?

Thank you!

**Comment by Frixionder** [comment]
Hello,

If i'm not mistaken, you can set a min and max value for any axis in options.plugins.zoom.limits.axisID.

In your case, it'd be options.plugins.zoom.limits.x.min and options.plugins.zoom.limits.x.max.
According to the documentation, the value can either be a number or "original', which _"instructs chartjs-plugin-zoom to use whatever limits the scale had when the chart was first displayed."_

```
chartConfig = {
  type:'line',
  data: //whatever you want
  options:{
    plugins:{
      zoom:{
        limits:{
          x:{
            min: // Min value
            max: // Max value
          }
        }
      }
    }
  }
}
```

[Link to the above-mentioned documentation](https://www.chartjs.org/chartjs-plugin-zoom/latest/guide/options.html#scale-limits)
